doc/Manuals/LoadBalancing/ig-lb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,738 @@
+| \[IG-SS\] | [X-Road: Security Server Installation Guide](../ig-ss_x-road_v6_security_server_installation_guide.md) |
+| \[UG-SS\] | [X-Road 6 Security Server User Guide](../) |
+
+
+## 2. Overview
+
+This document describes the external load balancing support features implemented by X-Road and the steps necessary to
+configure security servers to run as a cluster where each node has an identical configuration, including their keys and
+certificates. X-Road security server configuration changes are handled by a single master server and one or more slave
+servers.
+
+### 2.1 Goals and assumptions
+
+The primary goal of the load balancing support is, as the name suggests, load balancing, not fault tolerance.
+A clustered environment increases fault tolerance but some X-Road messages can still be lost if a security server node fails.
+
+The implementation does not include a load balancer component. It should be possible to use any external load balancer
+component that supports HTTP-based health checks for the nodes and load balancing at the TCP level (eg. haproxy, nginx,
+AWS ELB or Classic Load Balancing, or a hardware appliance). A health check service is provided for monitoring a node's
+status, this is described in more detail in section [3.4 Health check service configuration](#34-health-check-service-configuration)
+
+The load balancing support is implemented with a few assumptions about the environment that users should be aware of.
+Carefully consider these assumptions before deciding if the supported features are suitable for your needs.
+
+<a name="basic_assumptions"></a>
+__Basic Assumptions about the load balanced environment:__
+* Adding or removing nodes to or from the cluster is infrequent. New nodes need to be added manually and this takes some
+  time.
+* Changes to the configuration files are relatively infrequent and some downtime in ability to propagate the changes can
+  be tolerated. The cluster uses a master-slave model and the configuration master is not replicated.
+
+__Consequences of the selected implementation model:__
+* Changes to the `serverconf` database, authorization and signing keys are applied via the configuration master, which is
+  a member of the cluster. The replication is one-way from master to slaves and the slaves should treat the configuration
+  as read-only.
+* The cluster nodes can continue operation if the master fails but the configuration can not be changed until:
+  - the master comes back online, or
+  - some other node is manually promoted to be the master.
+* If a node fails, the messages being processed by that node are lost.
+  - It is the responsibility of the load balancer component to detect the failure and route further messages to other nodes.
+    Because there potentially is some delay before the failure is noticed, some messages might be lost due to the delay.
+  - Recovering lost messages is not supported by the load balancing support.
```

Comment:
maybe "by the load balancing solution" or something, to not have "supported....by the support"